agent/src/test/groovy/nu/studer/teamcity/buildscan/agent/AutoApplicationTest.groovy

```diff
@@ -0,0 +1,201 @@
+
+        given:
+        settingsFile << ''
+        buildFile << ''
+
+        when:
+        def jvmArgs = generateJvmArgs(mockScansServer.address, GE_VERSION, CCUD_VERSION)
+        def result = run(jdkCompatibleGradleVersion.gradleVersion, jvmArgs)
+
+        then:
+        outputContainsTeamCityServiceMessageBuildScanUrl(result)
+
+        where:
+        jdkCompatibleGradleVersion << SUPPORTED_GRADLE_VERSIONS
+    }
+
+    def "sends build scan url service message when GE and CCUD plugins are applied by project init script (#jdkCompatibleGradleVersion)"() {
+        assumeTrue jdkCompatibleGradleVersion.isJvmVersionCompatible()
+
+        given:
+        settingsFile << maybeAddGradleEnterprisePlugin(jdkCompatibleGradleVersion.gradleVersion)
+        buildFile << maybeAddBuildScanPlugin(jdkCompatibleGradleVersion.gradleVersion)
+
+        when:
+        def jvmArgs = generateJvmArgs(mockScansServer.address, GE_VERSION, CCUD_VERSION)
+        def result = run(jdkCompatibleGradleVersion.gradleVersion, jvmArgs)
+
+        then:
+        outputContainsTeamCityServiceMessageBuildScanUrl(result)
+
+        where:
+        jdkCompatibleGradleVersion << SUPPORTED_GRADLE_VERSIONS
+    }
+
+    @Ignore(value = "This behavior is desired but not yet implemented")
+    def "degrades gracefully project applies GE and init script applies CCUD <= 1.6.5 (#jdkCompatibleGradleVersion)"() {
+        assumeTrue jdkCompatibleGradleVersion.isJvmVersionCompatible()
+
+        given:
+        settingsFile << maybeAddGradleEnterprisePlugin(jdkCompatibleGradleVersion.gradleVersion)
+        buildFile << maybeAddBuildScanPlugin(jdkCompatibleGradleVersion.gradleVersion)
+
+        when:
+        def jvmArgs = generateJvmArgs(mockScansServer.address, GE_VERSION, '1.6.5')
+        def result = run(jdkCompatibleGradleVersion.gradleVersion, jvmArgs)
+
+        then:
+        outputContainsTeamCityServiceMessageBuildScanUrl(result)
+
+        where:
+        jdkCompatibleGradleVersion << SUPPORTED_GRADLE_VERSIONS
+    }
+
+    def "build succeeds when URL is set without applied GE plugin (#jdkCompatibleGradleVersion)"() {
+        assumeTrue jdkCompatibleGradleVersion.isJvmVersionCompatible()
+
+        given:
+        settingsFile << ''
+        buildFile << ''
+
+        when:
+        def jvmArgs = generateJvmArgs(mockScansServer.address, null, null)
+
+        then:
+        run(jdkCompatibleGradleVersion.gradleVersion, jvmArgs)
+
+        where:
+        jdkCompatibleGradleVersion << SUPPORTED_GRADLE_VERSIONS
+    }
+
+    def "build doesn't apply CCUD is applied without GE plugin (#jdkCompatibleGradleVersion)"() {
+        assumeTrue jdkCompatibleGradleVersion.isJvmVersionCompatible()
+
+        given:
+        settingsFile << ''
+        buildFile << ''
+
+        when:
+        def jvmArgs = generateJvmArgs(mockScansServer.address, null, CCUD_VERSION)
+
+        then:
+        if (jdkCompatibleGradleVersion.gradleVersion < GradleVersion.version('5.2')) {
```

Comment:
The behavior of applying CCUD without a Build Scan/GE plugin applied seems to be different for Gradle < 5.2 and Gradle >= 5.2